File: weather_mcp_sse.py
import uvicorn
from weather_mcp import mcp

# Tworzymy osobną aplikację używającą Server-Sent Events (SSE)
# To pozwala na korzystanie z serwera w przeglądarce i innych narzędziach po HTTP.
app = mcp.sse_app()

if __name__ == "__main__":
    print("Uruchamianie serwera pogody MCP (tryb SSE) na porcie 3000...")
    print("Dostęp przez http://localhost:3000/sse")
    # Uruchamiamy aplikację uvicorn tak, by można ją było podpiąć jako serwer http/sse
    uvicorn.run("weather_mcp_sse:app", host="0.0.0.0", port=3000)

File: typescript/src/weather_mcp_sse.ts
import express from "express";
import { SSEServerTransport } from "@modelcontextprotocol/sdk/server/sse.js";
import { createWeatherServer } from "./server.js";

const app = express();
const port = 3000;

const server = createWeatherServer();
let transport: SSEServerTransport;

app.use(express.json());

app.get("/sse", async (req, res) => {
    transport = new SSEServerTransport("/messages", res);
    await server.connect(transport);
});

app.post("/messages", async (req, res) => {
    if (transport) {
        await transport.handlePostMessage(req, res);
    } else {
        res.status(400).send("No active SSE connection");
    }
});

app.listen(port, () => {
    console.log(`Uruchamianie serwera pogody MCP (tryb SSE) na porcie ${port}...`);
    console.log(`Dostęp przez http://localhost:${port}/sse`);
});
